robot/wrapper.py

```diff
@@ -80,29 +83,28 @@ def __init__(self,
         # check if copy stat file exists and read it if it does then delete it
         try:
             with open(COPY_STAT_FILE, "r") as f:
-                _logger.info("Robot code copied %s from USB\n",
-                             f.read().strip())
+                _logger.info("Robot code copied %s from USB\n", f.read().strip())
             os.remove(COPY_STAT_FILE)
         except IOError:
             pass
 
-        self.subsystem_init(camera)
+        self.subsystem_init(camera, start_enable_12v, start_enable_5v)
         self.report_hardware_status()
-        self.enable_12v = True
         type(self)._initialised = True
 
         # Allows for the robot object to be set up and mutated before being set
         # up. Dangerous as it means the start info might not get loaded
         # depending on user code.
         if wait_for_start is True:
             start_data = self.wait_start()
-            self.zone = start_data or self.zone
+            self.zone = start_data['zone']
+            self.mode = start_data['mode']
         else:
             _logger.warning("Robot initalized but usercode running before"
-                            "`robot.wait_start`. Robot will not wait for the "
-                            "start button until `robot.wait_start` is called.")
+                           "`robot.wait_start`. Robot will not wait for the "
+                           "start button until `robot.wait_start` is called.")
 
-    def subsystem_init(self, camera):
+    def subsystem_init(self, camera, start_enable_12v, start_enable_5v):
```

Comment:
Change of interface here. Called on line 91 and hopefully has been updated anywhere else